tracing/src/main/java/com/palantir/tracing/CloseableTracer.java

```diff
@@ -17,13 +17,14 @@
 package com.palantir.tracing;
 
 import com.palantir.tracing.api.SpanType;
+import java.util.Map;
 
 /**
  * Wraps the {@link Tracer} methods in a closeable resource to enable the usage of the try-with-resources pattern.
  *
  * <p>Usage: try (CloseableTracer trace = CloseableTracer.start("traceName")) { [...] }
  */
-public final class CloseableTracer implements AutoCloseable {
+public class CloseableTracer implements AutoCloseable {
```

Comment:
Despite not being final, the private ctor ensures that this cannot be extended.

Reply:
heh, it's a little wonky. Our FinalClass errorprone rule has to check for this case https://github.com/palantir/gradle-baseline/blob/0f716d5ec6063b4dae06b5f761f252178aac3941/baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/FinalClass.java#L100-L102